brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/plan/ToscaTypePlanTransformer.java

```diff
@@ -67,7 +67,7 @@ public ToscaTypePlanTransformer() {
     }
 
     @Override
-    public void setManagementContext(ManagementContext managementContext) {
+    public synchronized void setManagementContext(ManagementContext managementContext) {
```

Comment:
can you explain the synch model?  there is an ugly *class* synch lock in `initialiseAlien` -- probably not a problem as we only want to initialise one but confusing to call it this method also holds an *instance* lock.  maybe use a `synchronized (this) { ... }` block here instead and move the `initialiseAlien` call to be outside it (using a boolean to record the fact that `this.mgmt` was set).  (avoid holding multiple locks)